libsolidity/ast/Types.cpp

```diff
@@ -340,110 +367,254 @@ MemberList::MemberMap IntegerType::nativeMembers(ContractDefinition const*) cons
+		_operator == Token::Dec ||
+		_operator == Token::After
+	)
+		return shared_from_this();
+	else
+		return TypePointer();
+}
+
+bool FixedPointType::operator==(Type const& _other) const
+{
+	if (_other.category() != category())
+		return false;
+	FixedPointType const& other = dynamic_cast<FixedPointType const&>(_other);
+	return other.m_integerBits == m_integerBits && other.m_fractionalBits == m_fractionalBits && other.m_modifier == m_modifier;
 }
 
-IntegerConstantType::IntegerConstantType(Literal const& _literal)
+string FixedPointType::toString(bool) const
 {
-	m_value = bigint(_literal.value());
+	string prefix = isSigned() ? "fixed" : "ufixed";
+	return prefix + dev::toString(m_integerBits) + "x" + dev::toString(m_fractionalBits);
+}
+
+TypePointer FixedPointType::binaryOperatorResult(Token::Value _operator, TypePointer const& _other) const
+{
+	if (
+		_other->category() != Category::RationalNumber 
+		&& _other->category() != category()
+		&& _other->category() != Category::Integer
+	)
+		return TypePointer();
+	auto commonType = dynamic_pointer_cast<FixedPointType const>(Type::commonType(shared_from_this(), _other));
+
+	if (!commonType)
+		return TypePointer();
+
+	// All fixed types can be compared
+	if (Token::isCompareOp(_operator))
+		return commonType;
+	if (Token::isBitOp(_operator) || Token::isBooleanOp(_operator))
+		return TypePointer();
+	return commonType;
+}
 
-	switch (_literal.subDenomination())
+tuple<bool, rational> RationalNumberType::isValidLiteral(Literal const& _literal)
+{
+	rational x;
+	try
 	{
-	case Literal::SubDenomination::Wei:
-	case Literal::SubDenomination::Second:
-	case Literal::SubDenomination::None:
-		break;
-	case Literal::SubDenomination::Szabo:
-		m_value *= bigint("1000000000000");
-		break;
-	case Literal::SubDenomination::Finney:
-		m_value *= bigint("1000000000000000");
-		break;
-	case Literal::SubDenomination::Ether:
-		m_value *= bigint("1000000000000000000");
-		break;
-	case Literal::SubDenomination::Minute:
-		m_value *= bigint("60");
-		break;
-	case Literal::SubDenomination::Hour:
-		m_value *= bigint("3600");
-		break;
-	case Literal::SubDenomination::Day:
-		m_value *= bigint("86400");
-		break;
-	case Literal::SubDenomination::Week:
-		m_value *= bigint("604800");
-		break;
-	case Literal::SubDenomination::Year:
-		m_value *= bigint("31536000");
-		break;
+		rational numerator;
+		rational denominator(1);
+		
+		auto radixPoint = find(_literal.value().begin(), _literal.value().end(), '.');		
+
+		if (radixPoint != _literal.value().end())
+		{
+			if (
+				!all_of(radixPoint + 1, _literal.value().end(), ::isdigit) || 
+				!all_of(_literal.value().begin(), radixPoint, ::isdigit)
+			)
+				throw;
+			//Only decimal notation allowed here, leading zeros would switch to octal.
+			auto leadingZeroes = find_if_not(
+				radixPoint + 1, 
+				_literal.value().end(), 
+				[](char const& a) { return a == '0'; }
+			);
+			auto fractionalBegin = leadingZeroes != _literal.value().end() ?
```

Comment:
It's possible to remove this condition, but I noticed something very odd when I was testing out the var capabilities...only the first rational type registers...everything after that does not register atleast in terms of a isValid literal...it does not hit that function. This should require some investigation or atleast some explanation as to why this might be.